documents/model.go

```diff
@@ -328,13 +341,13 @@ func (m *CoreDocumentModel) GetDocumentRootTree() (tree *proofs.DocumentTree, er
 	tree = NewDefaultTree(ConvertToProofSalts(document.CoredocumentSalts))
 
 	// The first leave added is the signing_root
-	err = tree.AddLeaf(proofs.LeafNode{Hash: document.SigningRoot, Hashed: true, Property: NewLeafProperty(SigningRootField, compactProperties[SigningRootField])})
+	err = tree.AddLeaf(proofs.LeafNode{Hash: document.SigningRoot, Hashed: true, Property: NewLeafProperty(SigningRootField, compactProperties(SigningRootField))})
 	if err != nil {
 		return nil, err
 	}
 
 	// For every signature we create a LeafNode
-	sigProperty := NewLeafProperty(SignaturesField, compactProperties[SignaturesField])
+	sigProperty := NewLeafProperty(SignaturesField, compactProperties(SignaturesField))
```

Comment:
Wouldn't it be possible to call flatten with `document.Signatures` here instead of iterating through them in line 364?

Reply:
Or will you implement this in a separate PR?

Reply:
Im not sure how this would work, let me check. I think it makes sense to do the changes in another pr together with moving stuff out of the repo. Trying to keep changes to a minimum so that we can move ahead with NFTs.

Reply:
OK, I'm fine with that. It should work out of the box, you can just create a tree from the signatures message.